onnxruntime/core/framework/tunable.h

```diff
@@ -215,9 +215,24 @@ class TunableOp {
     }
   }
 
-  static double Profile(Op<ParamsT>& op, const ParamsT* param) {
+  static double Profile(Op<ParamsT>& op, const ParamsT* param, double min_time, bool early_stop) {
     constexpr const int num_iter = 100;
+    constexpr const int early_stop_num_iter = 10;
+    constexpr const double early_stop_factor = 1.2;
     TimerT timer{param->Stream()};
+
+    if (early_stop) {
+      timer.Start();
+      for (int i = 0; i < early_stop_num_iter; i++) {
+        ORT_THROW_IF_ERROR(op(param));
+      }
+      timer.End();
+      double cur_time = timer.Duration() / early_stop_num_iter;
+      if (cur_time > min_time * early_stop_factor) {
```

Comment:
I think this logic belongs in `FindFastestImpl`. Perhaps have `Profile` accept a `num_iters` argument, which might be better than having to pass in the `min_time` and `early_stop` arguments.